libraries/stdlib/samples/test/samples/collections/maps.kt

```diff
@@ -104,7 +104,104 @@ class Maps {
                                          // prints: meal - 12.4
                                          // prints: dessert - 5.8
             }
+        }
+
+    }
+
+    class Filtering {
+
+        @Sample
+        fun filterKeys() {
+            val originalMap = mapOf("key1" to 1, "key2" to 2, "something_else" to 3)
+
+            val filteredMap = originalMap.filterKeys { it.contains("key") }
+            assertPrints(filteredMap, "{key1=1, key2=2}")
+            // original map has not changed
+            assertPrints(originalMap, "{key1=1, key2=2, something_else=3}")
+
+            val nonMatchingPredicate: (String) -> Boolean = { it == "key3" }
+            val emptyMap = originalMap.filterKeys(nonMatchingPredicate)
+            assertPrints(emptyMap, "{}")
+        }
+
+        @Sample
+        fun filterValues() {
+            val originalMap = mapOf("key1" to 1, "key2" to 2, "key3" to 3)
+
+            val filteredMap = originalMap.filterValues { it >= 2 }
+            assertPrints(filteredMap, "{key2=2, key3=3}")
+            // original map has not changed
+            assertPrints(originalMap, "{key1=1, key2=2, key3=3}")
+
+            val nonMatchingPredicate: (Int) -> Boolean = { it == 0 }
+            val emptyMap = originalMap.filterValues(nonMatchingPredicate)
+            assertPrints(emptyMap, "{}")
+        }
+
+        @Sample
+        fun filterTo() {
+            val originalMap = mapOf("key1" to 1, "key2" to 2, "key3" to 3)
+            val destinationMap = mutableMapOf("key40" to 40, "key50" to 50)
+
+            val filteredToMap = originalMap.filterTo(destinationMap) { it.value < 3 }
+
+            //destination map is updated with filtered items from the original map
+            assertEquals(destinationMap, filteredToMap)
```

Comment:
Better to assert `assertTrue(destinationMap === filteredMap)` to emphasize it's the same instance. Same in `filterNotTo` example.

Reply:
Thank you for your feedback! I have updated samples. Will add a couple of more samples if I have some free time next week :)